### PR TITLE
 Add ball effects, even for invalid projectiles

### LINKED ISSUE #3323: Particle traces are not always drawn after 15.1.3
Today I've found out that some players of my DDNet code based mod (infNext) with new DDNet client versions doesn't see some custom entities in my game build up from particle traces. The latest released version that displays them correctly is 15.1.3.

Bisecting on this repository I've found out that commit that changed the behavior was this one:

https://github.com/ddnet/ddnet/commit/8d6148b20b2cde94d8e1fbe0a0dcd0bd81e86acc

Looking in my code I've found that most custom entities that are not drawn correctly uses some trick borrowed from infClass:

```
		CNetObj_Projectile * pObj = static_cast<CNetObj_Projectile*>(Server()->SnapNewItem(NETOBJTYPE_PROJECTILE, m_IDs[NUM_SIDE + i], sizeof(CNetObj_Projectile)));
....
			pObj->m_Type = WEAPON_HAMMER;
```

So when type of CNetObj_Projectile is set to WEAPON_HAMMER or WEAPON_NINJA, the projectile itself were not drawn in teeworlds 0.6/0.7.5 and DDNet prior to 5.2. Instead trace particles were drawn, but now they're not.

There is no adequate substitute to that because setting m_Type to WEAPON_GUN, WEAPON_SHOTGUN or WEAPON_LASER would not only draw a particle that should not be there but will produce a very unpleasant flickering effect as it's position is updated serverside with every snapshot.

This technique were used in my mod (and in infCroya mod too) to draw Infection Circles slowly rotating contours, poison gas random particles and air bubble particles. I cannot think of any solution I could apply server-side to maintain the prior aesthetics and functionality for new DDNet client users.

So if that possible could you revert behavior to the old one? To draw a particle trace even if no particle itself is defined for a weapon. Thanks!

UPD: as this code is similar to infClass one, it affects old infClass servers gameplay too. 

image

image

image

cc: @Jupeyy

### PR DESCRIPTION
**Pull request by Jupeyy**
@pure-luck-999 can you test, if this is already enough?

fixes #3323

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/items.cpp [hunk 1]
```diff
@@ -24,20 +24,22 @@ void CItems::OnReset()
 
 void CItems::RenderProjectile(const CNetObj_Projectile *pCurrent, int ItemID)
 {
+	int CurWeapon = clamp(pCurrent->m_Type, 0, NUM_WEAPONS - 1);
```
Comment (heinrich5991) [inline src/game/client/components/items.cpp:27]: Please add a comment why you're doing what you're doing: Backward compatibility with certain mods.

Comment (Jupeyy) [reply src/game/client/components/items.cpp:27]: added it right before the if blocks of where the effects are added

**Comment by heinrich5991** [comment]
bors r+

Thanks. :)

**Comment by def-** [comment]
@Jupeyy @pure-luck-999  This should be released as 15.2.3 right now, correct?

**Comment by Jupeyy** [comment]
Yes

**Pull request merged**